Remove preprocessing of CFLAGS, CPPFLAGS, and LDFLAGS in Python scripts.

xuhdev opened the pull request:
After #23455, there is no need of this preprocessing in Python scripts.
They will be automatically processed in CMake (plus CPPFLAGS here
probably meant to be CXXFLAGS).

Reference:

- https://cmake.org/cmake/help/v3.15/envvar/CFLAGS.html
- https://cmake.org/cmake/help/v3.15/envvar/CXXFLAGS.html
- https://cmake.org/cmake/help/v3.15/envvar/LDFLAGS.html

xuhdev committed:
Remove preprocessing of CFLAGS, CPPFLAGS, and LDFLAGS in Python scripts.

After #23455, there is no need of this preprocessing in Python scripts.
They will be automatically processed in CMake (plus CPPFLAGS here
probably meant to be CXXFLAGS).

peterjc123 reviewed (commented):
Few questions:
1. I switched the docs to the version below 3.10 and they are gone. So are they available in legacy CMake versions?
2. Have you eliminated the usage of the `CMAKE_*` variables and renamed  `CPPFLAGS` to `CXXFLAGS` in the builder repo, too?

xuhdev commented:
>     1. I switched the docs to the version below 3.10 and they are gone. So are they available in legacy CMake versions?

At least since v3.5.0 they've existed. The document was missing before and added in https://gitlab.kitware.com/yrHeTaTeJlb/cmake/commit/e6b77c5f1a918b3b33457eca0590d0ff3840c08f

in CMake 3.5.0, `CFLAGS` and `LDFLAGS` were already used:

- https://gitlab.kitware.com/yrHeTaTeJlb/cmake/blob/v3.5.0/Modules/CMakeCInformation.cmake#L113
- https://gitlab.kitware.com/yrHeTaTeJlb/cmake/blob/v3.5.0/Modules/CMakeCommonLanguageInclude.cmake#L19-23

>     2. Have you eliminated the usage of the `CMAKE_*` variables and renamed  `CPPFLAGS` to `CXXFLAGS` in the builder repo, too?

`CPPFLAGS` is not used in the builder repo and this commit did not change the behavior of `CXXFLAGS`.

suo commented:
It looks like this PR broke asan builds ([example](https://circleci.com/gh/pytorch/pytorch/2320783?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link/console)). @xuhdev is this a quick fix? Otherwise we may have to revert.

xuhdev commented:
@suo I'm not sure whether there is a quick fix. Feel free to revert for now if you believe this patch is the reason (for which I haven't seen why).

suo commented:
@xuhdev not sure why this would cause the problem; I'm just going by https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master, which shows that the asan builds started failing after this commit was applied + the same build failed on this PR as well.

xuhdev commented:
@suo I came up with a potential quick fix. Hold on one moment
